docs/admin/external-auth.md

```diff
@@ -62,6 +62,55 @@ Use [`external-auth`](../reference/cli/external-auth.md) in the Coder CLI to acc
 coder external-auth access-token <USER_DEFINED_ID>
 ```
 
+## Git Authentication in Workspaces
+
+Coder provides automatic Git authentication for workspaces through SSH authentication and Git-provider specific env variables.
+
+When performing Git operations, Coder first attempts to use external auth provider tokens if available.
+If no tokens are available, it defaults to SSH authentication.
+
+### OAuth (external auth)
+
+For Git providers configured with [external authentication](#configuration), Coder can use OAuth tokens for Git operations.
+
+When Git operations require authentication, and no SSH key is configured, Coder will automatically use the appropriate external auth provider based on the repository URL.
+
+For example, if you've configured a GitHub external auth provider and attempt to clone a GitHub repository, Coder will use the OAuth token from that provider for authentication.
+
+To manually access these tokens within a workspace:
+
+```shell
+coder external-auth access-token <USER_DEFINED_ID>
+```
+
+### SSH Authentication
```

Comment:
I feel like that this whole SSH authentication section should not be here. 
It should probably go to admin/users. 

I don't have a strong opinion so will leave this to you.